packages/platforms/accton/arm64/as4224/src/modules-5_10/arm64-accton-as4224-psu.c

```diff
@@ -0,0 +1,265 @@
+/*
+ * An hwmon driver for accton as4224 Power Module
+ *
+ * Copyright (C) 2014 Accton Technology Corporation.
+ * Brandon Chuang <brandon_chuang@accton.com.tw>
+ *
+ * Based on ad7414.c
+ * Copyright 2006 Stefan Roese <sr at denx.de>, DENX Software Engineering
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+
+#include <linux/module.h>
+#include <linux/jiffies.h>
+#include <linux/hwmon.h>
+#include <linux/hwmon-sysfs.h>
+#include <linux/err.h>
+#include <linux/mutex.h>
+#include <linux/sysfs.h>
+#include <linux/slab.h>
+#include <linux/delay.h>
+#include <linux/dmi.h>
+#include <linux/platform_device.h>
+
+#define DRVNAME "as4224_psu"
+
+#define BOARD_INFO_REG_OFFSET       0x00
+#define PSU_STATUS_I2C_ADDR         0x40
+#define PSU_STATUS_I2C_REG_OFFSET   0x03
+
+#define IS_POWER_GOOD(id, value)    (!!(value & BIT(id + 1)))
+
+static ssize_t show_count(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t show_status(struct device *dev, struct device_attribute *da, char *buf);
+extern int as4224_cpld_read(unsigned short cpld_addr, u8 reg);
+
+/* Each client has this additional data
+ */
+struct as4224_psu_data {
+    struct platform_device *pdev;
+    struct device      *hwmon_dev;
+    struct mutex       update_lock;
+    char               valid;        /* !=0 if registers are valid */
+    unsigned long       last_updated; /* In jiffies */
+    u8  status;          /* power_good status read from CPLD */
+    u8  psu_count;
```

Comment:
Would be nice if the Linux kernel coding style could be followed.

Reply:
Modified to follow Linux kernel coding style.